history.go

```diff
@@ -46,24 +69,24 @@ func (h *History) LookUpInEpoch(key string, ep int64) (MerkleNode, *Authenticati
 	return n, ap, nil
 }
 
-func (h *History) GetSTR(ep int64) *SignedTreeRoot {
-	pointer := h.str
-	for pointer.epoch > ep && pointer != nil {
-		if pointer.prev == nil {
+func (h *History) GetSTR(ep Epoch) *SignedTreeRoot {
+	pointer := h.hashChain[h.currentSTR.epoch]
+	for pointer.epoch > ep && h.hashChain[pointer.epoch] != nil {
```

Comment:
The current implementation does more naturally traverse the hash chain, this is true. But I think taking advantage of having the ``epochs`` structure is a bit cleaner since there is one less level of indirection trying to get at the STR's epoch. On the other hand, using ``epochs`` as the key into ``hashChain`` will probably require us to confirm that ``str.epoch == h.epochs[i]`` once the appropriate STR in the chain has been selected.